drivers/bus/qspi.h

```diff
@@ -52,6 +52,22 @@ typedef struct _mp_soft_qspi_obj_t {
     mp_hal_pin_obj_t io3;
 } mp_soft_qspi_obj_t;
 
+#define MP_SPI_ADDR_32B(addr) (addr & 0xF000)
```

Comment:
Shouldn't this be `(addr & 0xff000000)`?  Since `addr` is measured in bytes everywhere (I think, that would need to be verified).

Reply:
Wow... I'm, not sure now where the 0xF000 came from... unless in an earlier iteration of this change that val was in blocks?

Either way, yes I agree with the change. It also likely explains the corruption issue I've posted about.

Thanks for picking this up!